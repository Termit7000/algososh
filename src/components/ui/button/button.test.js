import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'

import { Button } from './button';

describe("Button", () => {


    it("Кнопка с текстом", () => {
        const button = <Button text='some text'/>;
        render(button);
        expect(screen.getByTestId('button')).toMatchSnapshot();
    })
    
    it("Кнопка без текста", () => {
        const button = <Button/>;
        render(button);
        expect(screen.getByTestId('button')).toMatchSnapshot();
    })

    it("Заблокированная кнопка", () => {
        const button = <Button disabled={true}/>;
        render(button);
        expect(screen.getByTestId('button')).toMatchSnapshot();
    })

    it("Кнопка с индикацией загрузки", () => {
        const button = <Button isLoader={true}/>;
        render(button);
        expect(screen.getByTestId('button')).toMatchSnapshot();
    })

    it('Корректность вызова колбека при клике на кнопку', ()=>{
        const callBack = jest.fn();

        const button = <Button onClick={callBack}/>;

        render(button);

        const element = screen.getByTestId('button');
        expect(element).toBeInTheDocument();
        
        fireEvent.click(element);
        expect(callBack).toBeCalled();
    })
})