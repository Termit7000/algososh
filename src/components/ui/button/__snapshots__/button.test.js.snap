// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Заблокированная кнопка 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button Кнопка без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button Кнопка с индикацией загрузки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Кнопка с текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  >
    some text
  </p>
</button>
`;
